pkg/reconciler/internal/hook/hook_test.go

```diff
@@ -127,8 +124,6 @@ var _ = Describe("Hook", func() {
 				drw = internalhook.NewDependentResourceWatcher(c, rm, cache, sch)
 				Expect(drw.Exec(owner, *rel, log)).To(Succeed())
 				Expect(c.WatchCalls).To(HaveLen(2))
-				Expect(c.WatchCalls[0].Handler).To(BeAssignableToTypeOf(handler.EnqueueRequestForOwner(sch, rm, owner, handler.OnlyControllerOwner())))
-				Expect(c.WatchCalls[1].Handler).To(BeAssignableToTypeOf(handler.EnqueueRequestForOwner(sch, rm, owner, handler.OnlyControllerOwner())))
```

Comment:
Removed because the `ctrl.Watch()` function signature changed to no longer accept handlers and predicated. They are instead configured via the `source.Source` type. Instead of creating a custom implementation of the `source.Source` interface to facilitate this unit test I felt it would be better to remove these assertions as I'm not entirely sure they are that valuable (although I could be wrong, so please let me know if I am!).

Reply:
It looks like the reason for the existence of these tests is to check whether the correct handler is used. If we are no longer checking the handler specifics, then these tests aren't checking what they say they're checking.

Reply:
Looks like the option to add additional handler and predicates was to allow passing them during `hook_test` (https://github.com/operator-framework/helm-operator-plugins/blob/ae087c419483d73c37155c02f142ab14b4d8c706/pkg/reconciler/internal/hook/hook_test.go#L174-L178). We may face issues with that test, if we remove this.

Reply:
I'll work on getting this done in a way that we can still have these assertions